cxx-qt-gen/src/parser/signals.rs

```diff
@@ -0,0 +1,145 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::parser::parameter::ParserParameter;
+use crate::syntax::fields::fields_named_to_ident_type;
+use syn::{Ident, ItemEnum, Result};
+
+/// Describes an individual Signal
+pub struct ParsedSignal {
+    /// The name of the signal
+    pub ident: Ident,
+    /// The parameters of the signal
+    pub parameters: Vec<ParserParameter>,
+    // TODO: later this will describe of the signal has an attribute
+    // stating that the signal exists in the base class
+}
+
+/// Describes a Signals block for a QObject
+pub struct ParsedSignals {
+    /// The name of the signals enum
+    pub ident: Ident,
+    /// The original enum for the signals
+    pub item: ItemEnum,
+    /// A list of the signals defined in the enum
+    pub signals: Vec<ParsedSignal>,
+}
+
+impl ParsedSignals {
+    /// Constructs a ParsedSignals object from a given [syn::ItemEnum] block
+    pub fn from(item: &ItemEnum, attr_index: usize) -> Result<Self> {
+        // Remove the attribute index as we have processed it
+        let mut item = item.clone();
+        item.attrs.remove(attr_index);
+
+        let signals = item
+            .variants
+            .iter()
+            .map(|variant| {
```

Comment:
The contenst of this lambda should probably be a constructor in `ParsedSignal`.
E.g. `ParsedSignal::from_variant`.

Reply:
Done